zips/zip-0231.md

```diff
@@ -0,0 +1,332 @@
+> ZIP: 231
+> Title: Memo Bundles
+> Owners: Jack Grigg <jack@electriccoin.co>
+>         Kris Nuttycombe <kris@electriccoin.co>
+>         Daira-Emma Hopwood <daira@electriccoin.co>
+>         Arya Solhi <arya@zfnd.org>
+> Credits: Sean Bowe
+>          Nate Wilcox
+> Status: Draft
+> Category: Consensus / Wallet
+> Created: 2024-04-26
+> License: MIT
+
+
+# Terminology
+
+{Edit this to reflect the key words that are actually used.}
+The key words "MUST", "REQUIRED", "MUST NOT", "SHOULD", and "MAY" in this
+document are to be interpreted as described in BCP 14 [#BCP14]_ when, and
+only when, they appear in all capitals.
+
+
+# Abstract
+
+TBD
+
+
+# Motivation
+
+In Zcash transaction versions v2-v5, each shielded output contains a ciphertext
+comprised of a 52-byte note plaintext, and a corresponding 512-byte memo field.
+Recipients can only decrypt the outputs sent to them, and thus can also only
+observe the memo fields included with the outputs they can decrypt.
+
+The shielded transaction protocol hides the sender(s) (that is, the addresses
+corresponding to the keys used to spend the input notes) from all of the
+recipients. For certain kinds of transactions, it is desirable to make one or
+more sender addresses available to one or more recipients (for example, a reply
+address), and in subsets of those transactions it is desirable to authenticate
+the sender addresses (to give the recipient a guarantee that the address is
+controlled by a sender of the transaction). These Authenticated Reply Addresses
+require zero-knowledge proofs, and for the Orchard protocol these proofs are too
+large to fit into a 512-byte memo field.
```

Comment:
Another use case we should be concerned with here is coin voting - can the proofs in Hanh's coin voting protocol (along with the amount being voted with) fit into a memo?